browser-test/src/application_review.test.ts

```diff
@@ -85,6 +86,10 @@ describe('normal application flow', () => {
       'monthly-income-q',
     ])
 
+    // Intentionally add an empty block to ensure that empty blocks do not
```

Comment:
> This existing test is gross.

why?

> I also feel like the use case is an edge case (EG not a conscious part of the design) compared to the rest of this test, that warrants its own test also.

> FWIW, I don't think ind test framework setup/teardown affects the test run timing as much as the tests interacting with the UI.

I don't agree, we saw a pretty noticeable improvement in test runtime when we merged a bunch of smaller browser tests into single longer ones. I'd prefer to leave it in that condition for now.

The best way to improve the situation overall IMO would be to split out multiple CI tasks for running different batches of the browser tests in parallel. With that in place it'd probably be a lot less painful to have a bunch of smaller, more focused browser tests.

Reply:
Gross:  It's doing a lot a lot of things which makes it unfocused as far as test coverage and confidence go as well readability, which makes updating it error prone.

What is "normal" and how would a new dev know if they should update "normal"
 
This test is also both verifying validation concerns and resolution concerns some questions, but it's hard to know what the coverage is from the method naming/results list. Those concerns should ideally be more unit ized so that one question failure doesn't affect all question verification status.

As for the blocks, it's unclear why the blocks are organized that way, such that the testing they are doing is easily lost through future changes.  Aside from "More than 1 block" I don't know why there's 5 blocks versus 2.


Test Timing:

Is it possible that the time was improved because of sharing the Admin/Program setup, and not the user interaction with the program itself?

EG: we can have separate tests, share the same Program setup.  I've seen where it's all or nothing and not the middle ground.

Reply:
Here's an example of sharing setup I did with the currency question type

https://github.com/seattle-uat/civiform/blob/6724de093b884d32af4ded5c38404fc84c25501d/browser-test/src/currency.test.ts#L47

Reply:
captured this in a tech debt item to address down the line